### PR TITLE
Add test with predicate on top level and nested field

### LINKED ISSUE #15393: Unable to query for iceberg table if there are same field names in top level and nested struct
We are using Trino 387.

I tried to use Iceberg connector, but some queries are failed with `Multiple entries with same key` error.

For verify the problem, I've created a table for test as below using spark sql:

```sql
-- HiveCatalog: dev

create table dev.test_db.iceberg_trino_test (
 id string,
 nested1 struct<id: string>,
 nested2 struct<id: string>
) using iceberg;

insert into dev.test_db.iceberg_trino_test (id, nested1, nested2)
 values ('0', named_struct('id', '1'), named_struct('id', '2'));
```

Then tried to query on Trino, and got error. (The query works well on Hive)

```sql
select * from test_db.iceberg_trino_test
where nested1.id='1' and nested2.id='2';
```

`Query failed (#20221214_094933_00231_ngidj): Error opening Iceberg split hdfs://warehouse/test_db.db/iceberg_trino_test/data/00000-0-9c521b08-195f-4246-bbec-064bc007fe65-00001.parquet (offset=4, length=1029): Multiple entries with same key: [id] required binary id (STRING) = 1=[ SortedRangeSet[type=varchar, ranges=1, {[2]}] ] and [id] required binary id (STRING) = 1=[ SortedRangeSet[type=varchar, ranges=1, {[1]}] ] java.lang.IllegalArgumentException: Multiple entries with same key: [id] required binary id (STRING) = 1=[ SortedRangeSet[type=varchar, ranges=1, {[2]}] ] and [id] required binary id (STRING) = 1=[ SortedRangeSet[type=varchar, ranges=1, {[1]}] ]`

I think that iceberg connector aware just last field name for get `field-id`, not including whole nested structure.

### PR DESCRIPTION
**Pull request by pajaks**
## Description
Add test with predicate on top level and nested field

## Additional context and related issues
Improves test coverage for fix #15408
Fixes #15393

## Release notes

( x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 1]
```diff
@@ -3839,6 +3839,18 @@ public void testPredicateOnRowTypeField()
         }
     }
 
+    @Test
+    public void testPredicateOnTwoRowsWithTheSameFieldNames()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT) && hasBehavior(SUPPORTS_ROW_TYPE));
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_predicate_on_two_rows", "(row1 row(x INT), row2 row(x INT))")) {
```
Comment (ebyhr) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:3847]: Please make sure this test fails without #15408 in Iceberg connector using Parquet format. I'm sure it passes even without the commit.

**Review by ebyhr — COMMENTED** [review]
Please remove the following dot from the commit title.https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

**Comment by ebyhr** [comment]
Could you add the below comment to PR description so that we can close the linked issue automatically? 
```
Fixes #15393
```

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java (hunk 1 above)
Comment (bluzy) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:3847]: I think the table structure should be:

```suggestion
        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_predicate_on_two_rows", "(x INT, row1 row(x INT))")) {
```

Comment (pajaks) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:3847]: @bluzy I think case you are proposing is covered in testPredicateOnRowTypeField. Isn't it?

Comment (bluzy) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:3847]: How about this query into `testPredicateOnRowTypeField` ?

`assertQuery("SELECT int_t FROM " + table.getName() + " WHERE int_t = 2 AND row_t.int_t = 1", "VALUES 1");`

**Review by ebyhr — APPROVED** [review]

**Pull request merged**